crates/anstyle-roff/src/lib.rs

```diff
@@ -0,0 +1,166 @@
+            &mut doc,
+        );
+        set_effects_and_text(&styled, &mut doc);
+    }
+    doc
+}
+
+fn set_effects_and_text(styled: &StyledStr, doc: &mut Roff) {
+    // Roff (the crate) only supports these inline commands
+    //  - Bold
+    //  - Italic
+    //  - Roman (plain text)
+    // If we want more support, or even support combined formats, we will need
+    // to push improvements to roff upstream or implement a more thorough roff crate
+    // perhaps by spinning off some of this code
+    let effects = styled.style.get_effects();
+    if effects.contains(anstyle::Effects::BOLD) | has_bright_fg(&styled.style) {
+        doc.text(vec![bold(styled.text)]);
+    } else if effects.contains(anstyle::Effects::ITALIC) {
+        doc.text(vec![italic(styled.text)]);
+    } else {
+        doc.text(vec![roff::roman(styled.text)]);
+    }
+}
+
+fn has_bright_fg(style: &Style) -> bool {
+    style
+        .get_fg_color()
+        .as_ref()
+        .map(is_bright)
+        .unwrap_or(false)
+}
+
+/// Check if Color is an AnsiColor::Bright* variant
+fn is_bright(fg_color: &Color) -> bool {
+    if let Color::Ansi(color) = fg_color {
+        matches!(
+            color,
+            AnsiColor::BrightRed
+                | AnsiColor::BrightBlue
+                | AnsiColor::BrightBlack
+                | AnsiColor::BrightCyan
+                | AnsiColor::BrightGreen
+                | AnsiColor::BrightWhite
+                | AnsiColor::BrightYellow
+                | AnsiColor::BrightMagenta
+        )
+    } else {
+        false
+    }
+}
+
+type ColorSet<'a> = (&'a Option<Color>, &'a Option<Color>);
+
+fn set_color(colors: ColorSet, doc: &mut Roff) {
+    add_color_to_roff(doc, control_requests::FOREGROUND, colors.0);
+    add_color_to_roff(doc, control_requests::BACKGROUND, colors.1);
+}
+
+fn add_color_to_roff(doc: &mut Roff, control_request: &str, color: &Option<Color>) {
+    match color {
+        Some(Color::Rgb(c)) => {
+            // Adding Support for RGB colors, however cansi does not support
+            // RGB Colors, so this is not executed. If we switch to a provider
+            // That has RGB support we will also get it for Roff
+            let name = rgb_name(c);
+            doc.control(
+                control_requests::CREATE_COLOR,
+                vec![name.as_str(), "rgb", as_hex(c).as_str()],
+            )
+            .control(control_request, vec![name.as_str()]);
+        }
+
+        Some(Color::Ansi(c)) => {
+            doc.control(control_request, vec![ansi_color_to_roff(c)]);
+        }
+        Some(Color::XTerm(c)) => {
+            // Adding Support for XTerm colors, however cansi does not support
+            // XTerm Colors, so this is not executed. If we switch to a provider
+            // That has Xterm support we will also get it for Roff
+            let ansi_color = anstyle_lossy::xterm_to_ansi(*c, Palette::default());
```

Comment:
xterm is losslessly convertable to `Rgb | Ansi`.  Ideally, we'd do that.  Maybe some API changes are needed to make that easier.

Reply:
I used `anstyle_lossy::xterm_to_ansi` because It looked like it has the same functionality as the `XTermColor::into_ansi` function, but with the added benefit that additional values would get mapped to something _close_ to the requested color. Is there something I'm misunderstanding?

Reply:
Thats a lossy conversion which you don't need but can instead fo:

```rust
(*c).into_ansi().unwrap_or_else(|| anstyle_lossy::xterm_to_rgb(*c, Palette::default());
```
While this accepts a palette, it won't be used and  no lossy conversions are needed.  I'm mixed about adding a `XTerm::into_rgb` as the lookup tables are big but making this more clear its lossless (and not needing a lossy crarte for it) would be nice.

Reply:
Ahh, I think I see now. Rather than go straight to Ansi Colors (which would be lossy), use the roff RGB support for XTerm colors that don't map to Ansi, thereby getting all xterm colors supported.